src/models/cost_function_timeseries.jl

```diff
@@ -0,0 +1,83 @@
+function get_variable_cost(ts::IS.TimeSeriesData, ::MarketBidCost)
+    data = TimeSeries.TimeArray(TimeSeries.timestamp(ts), map(VariableCost, ts.data))
+    return data
+end
+
+function get_variable_cost(
+    device::Generator,
+    cost::MarketBidCost;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    time_series_key = get_variable(cost)
+    ts_contanier = IS.get_time_series_container(device)
+    raw_data = IS.get_time_series_values(
+        time_series_key.time_series_type,
+        device,
+        time_series_key.name,
+        start_time = start_time,
+        len = len,
+    )
+    time_stamps = IS.get_time_series_timestamps(
+        time_series_key.time_series_type,
+        device,
+        time_series_key.name,
+        start_time = start_time,
+        len = len,
+    )
+    return TimeSeries.TimeArray(time_stamps, map(VariableCost, raw_data))
+end
+
+function get_services_bid(
+    device::Generator,
+    cost::MarketBidCost,
+    service::Service;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    services_dict = get_ancillary_services(cost)
+    time_series_type = services_dict[get_name(service)]
+    raw_data = IS.get_time_series_values(
+        time_series_type,
+        device,
+        get_name(service),
+        start_time = start_time,
+        len = len,
+    )
+    time_stamps = IS.get_time_series_timestamps(
+        time_series_type,
+        device,
+        get_name(service),
+        start_time = start_time,
+        len = len,
+    )
+    return TimeSeries.TimeArray(time_stamps, map(VariableCost, raw_data))
+end
+
+function set_variable_cost!(
+    sys::System,
+    component::Generator,
```

Comment:
These functions should be on StaticInjectionDevice since Storage can participate and also loads.